RHDEVDOCS-7458: Content creation for Cluster resolver migration task (#112625)
openshift/openshift-docs: modules/op-migrate-custom-clustertasks.adoc

.Prerequisites

* You have identified the custom `ClusterTask` resources in your cluster.
* You have the necessary permissions to create `Task` resources.

.Procedure

. Export the existing `ClusterTask` resource to a YAML file by running the following command:
+
[source,terminal]
----
$ oc get clustertask my-custom-task -o yaml > my-custom-task.yaml
----

. Edit the YAML file to convert it to a namespace-scoped `Task`:
+
--
.. Change `kind: ClusterTask` to `kind: Task`.
.. Update `apiVersion` from `tekton.dev/v1beta1` to `tekton.dev/v1` if necessary.
.. Add a `namespace` field under `metadata`.
.. Remove generated fields such as `creationTimestamp`, `generation`, `resourceVersion`, `selfLink`, and `uid`.
.. Remove the `status` section if present.
--
+
--
*Example of a converted task:*
[source,yaml]
----
apiVersion: tekton.dev/v1
kind: Task
metadata:
  name: my-custom-task
  namespace: my-shared-tasks
spec:
  params:
  - name: MESSAGE
    type: string
    default: "Hello from custom task"
  steps:
  - name: echo-message
    image: registry.access.redhat.com/ubi9/ubi-minimal:latest
    script: |
      #!/bin/sh
      echo "$(params.MESSAGE)"
----

> `metadata.namespace`:: Specifies the namespace where the task will be created.

Reviewer: Nit:
```suggestion
`metadata.namespace`:: Specifies the namespace where the task is created.
```

Author: Fixed.

--

. Create the namespace if it does not exist by running the following command:
+
[source,terminal]
----
$ oc create namespace my-shared-tasks
----

. Apply the converted task to the cluster by running the following command:
+
[source,terminal]
----
$ oc apply -f my-custom-task.yaml
----

. Update all pipeline references to use the cluster resolver by running the following command:
+
[source,yaml]
----
apiVersion: tekton.dev/v1
kind: Pipeline
metadata:
  name: example-pipeline
spec:
  tasks:
  - name: run-custom-task
    taskRef:
      resolver: cluster
      params:
      - name: kind
        value: task
      - name: name
        value: my-custom-task
      - name: namespace
        value: my-shared-tasks 
----

> `spec.value`:: Specifies the namespace where you have created the converted task.

Reviewer: `value` is nested deeper:
```suggestion
`spec.taskRef.params.value`:: Specifies the namespace where you have created the converted task.
```
OR

```suggestion
`params.value`:: Specifies the namespace where you have created the converted task.
```

Author: Fixed.
